homeassistant/components/geo_location/__init__.py

```diff
@@ -0,0 +1,153 @@
+"""
+Geo Location component.
+
+This component covers platforms that deal with external events that contain
+a geo location related to the installed HA instance.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/geo_location/
+"""
+import logging
+from datetime import timedelta
+from typing import Optional
+
+from homeassistant.components import group
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE
+from homeassistant.helpers.entity import Entity, generate_entity_id
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA  # noqa
+from homeassistant import util
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SORT_GROUP_ENTRIES_REVERSE = False
+DOMAIN = 'geo_location'
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+GROUP_NAME_ALL_EVENTS = 'All Geo Location Events'
+SCAN_INTERVAL = timedelta(seconds=60)
+
+
+async def async_setup(hass, config):
+    """Setup this component."""
+    component = EntityComponent(
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL, GROUP_NAME_ALL_EVENTS)
+    await component.async_setup(config)
+    return True
+
+
+class GeoLocationDeviceManager:
```

Comment:
The reason why I added this is to give platforms in this component a way to manage their entities. In particular this manager would be used to group the entities, and to give the entities a common id prefix.
The GeoRSS platform that I had removed from this PR also uses a (more specific) manager to retrieve data from an external source and then generates multiple entities and needs to keep track of them.
Would you prefer to move all this code into the `demo` platform for now, and then for the next platform after this to either duplicate this code or move it back into a common base class?

Reply:
I don't think we should group entities by default. That's up to the user. There's the option to make a group platform for this component, if that makes sense.

The `entity_id` for entities of this component, will have as prefix the domain of this component `'geo_location'`. The rest of the id should be defined by the platform since the platform knows what can be an id for entities of that platform.

Each platform should add entities, via `add_devices`, and keep track of entities as needed.

Reply:
Alright, I'll move this into the `demo` platform for now.
Having already implemented the GeoRSS platform, I believe that automated grouping would be quite useful. The platform generates and removes entities all the time, and the user won't be able to manually group them.